src/Analysis/Ast/Impl/Specializations/BuiltinsSpecializations.cs

```diff
@@ -91,6 +93,79 @@ public static IMember List(IPythonInterpreter interpreter, IPythonFunctionOverlo
             return args.Count > 0 && args[0] is PythonCollection c ? c.Contents.FirstOrDefault() : null;
         }
 
+        public static IMember Super(IPythonModule declaringModule, IPythonFunctionOverload overload, IArgumentSet argSet, IndexSpan indexSpan) {
+            if (argSet.Arguments.Any()) {
+                // If multiple arguments first argument is required
+                var classType = argSet.Argument<IMember>(0).GetPythonType<IPythonClassType>();
+                if (classType?.Mro == null) {
+                    return null;
+                }
+
+                // second argument optional
+                bool isUnbound = argSet.Argument<IMember>(1) == null;
+                if (isUnbound) {
+                    return CreateSuper(argSet, classType.Mro);
+                } else {
+                    // Second argument can be either an object instance or a class which are both IPythonType
+                    IPythonType objOrType = null;
+                    objOrType = argSet.Argument<IMember>(1).GetPythonType();
+                    bool isInstance = objOrType?.Equals(classType) ?? false;
+
+                    if (!isInstance) {
+                        var argAsType = argSet.Argument<IMember>(1).GetPythonType<IPythonClassType>();
+                        bool isSubClass = classType?.Bases.Any(b => argAsType.Equals(b)) ?? false;
```

Comment:
is there a helper function for isSubClass?

Reply:
I added IsSubClassOf

Reply:
I think it should be `IsBaseOf` or `IsSuperClassOf` since base == superclass. Subclass is what is derived from the base (i.e narrowing operation). Cat is a subclass of an Animal. so Animal is base/superclass of Cat.